src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -152,7 +152,7 @@ private X509Certificate ServerCertSelectionCallbackWrapper(string targetHost)
 
         private SslAuthenticationOptions CreateAuthenticationOptions(SslServerAuthenticationOptions sslServerAuthenticationOptions)
         {
-            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null)
+            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null && _certSelectionDelegate == null)
             {
                 throw new ArgumentNullException(nameof(sslServerAuthenticationOptions.ServerCertificate));
             }
```

Comment:
We're missing check for ServerCertificateSelectionCallback & _certSelectionDelegate cannot be set both at the same time.

Reply:
The existing checks already cover that case, and throw:
```
System.InvalidOperationException : The 'ServerCertificateSelectionCallback' option was already set in the SslStream constructor.
```

Reply:
Actually, it looks like that isn't right. I based this check off the fact that all of the methods for starting a connection require a cert, but with the new SslServerAuthenticationOptions class that actually isn't the case. It looks like I do need to explicitly check for this case. I'll add that.

Reply:
I think the other check should be something like `if (newCertCallbackSet && (oldCertCallbackSet || ServerCertificatePropertySet)) error();` - previously we allowed ServerCertProp and callback now we want to still support that but error if you try to use new and old way at the same time

Reply:
Fixed in the latest commit. See the change to line 160.

Reply:
@kdg3737 it's interesting idea - it may make the code a bit clearer. Given we are nearly done with 2.1, we should keep changes to minimum at this point, so let's stick with what we have for now.